File: src/types/domain.ts
export interface User {
  name: string
  email: string
  enrichments: Enrichment[]
}

export interface Enrichment {
  companyPartner: string
  facultySupervisor: string
  location: string
  officePhoneNumber: string
  position: string
  semesterName: string
  siteSupervisorEmail: string
  siteSupervisorName: string
  siteSupervisorPhoneNumber: string
  track: string
  workingOfficeHours: string
}

export interface LoginParams {
  email: string
  password: string
}

File: scraper/internal/student/domain.go
package student

type Student struct {
	Name  string
	Email string
	// ProfilePictureURL string (do this later because base64 encoding)
	Enrichments []Enrichment
}

type Enrichment struct {
	SemesterName              string
	Track                     string
	CompanyPartner            string
	Position                  string
	Location                  string
	WorkingOfficeHours        string
	SiteSupervisorName        string
	SiteSupervisorEmail       string
	SiteSupervisorPhoneNumber string
	FacultySupervisor         string
	OfficePhoneNumber         string
}
